gammapy/time/lightcurve.py

```diff
@@ -112,3 +116,249 @@ def compute_fvar(self):
+        Returns list of OFF events for each observations
+        """
+        off_evt_list = []
+        for bg in spectral_extraction.background:
+            off_evt_list.append(bg.off_events)
+        return off_evt_list
+
+    
+    def _get_on_evt_list(self, spectral_extraction):
+        """
+        Returns list of OFF events for each observations
+        """
+        on_region = spectral_extraction.target.on_region
+        on_evt_list = []
+        for obs in spectral_extraction.obs:
+            idx = on_region.contains(obs.events.radec)
+            on_evt_list.append(obs.events.select_row_subset(idx))
+            
+        return on_evt_list
+    
+            
+    def _get_time_intervals(self, t_start, t_stop, time_binning_type):
+        """
+        Compute time intervals needed for a light curve
+
+        Parameters
+        ----------
+        time_start: `~astropy.units.Quantity`
+            start time to integrate events
+        time_stop: `~astropy.units.Quantity`
+            start time to integrate events
+        time_binning_type: `str`
+            type of time binning (e.g. obs, minute, hour, night, day, month, user)
+        """
+
+        dt = None
+        n_dt = None
+        intervals = []
+        if time_binning_type == 'obs':
+            n_dt = len(self.obs_list)
+            for obs in self.obs_list:
+                # for now use time of first and last evt to define the interval
+                # intervals.append([obs.tstart, obs.tstop])
+                intervals.append([obs.events.time[0], obs.events.time[-1]])
+        else:
+            raise NotImplementedError
+
+        return intervals
+
+    
+    def light_curve(self, t_start, t_stop, t_binning_type,
+                    spectral_model, energy_range, binning=None):
+        """
+        Function returning light curve for given parameters
+        
+        Parameters
+        ----------
+        t_start: `~astropy.units.Quantity`
+            start time to integrate events
+        t_stop: `~astropy.units.Quantity`
+            start time to integrate events
+        t_binning_type: `str`
+            type of time binning (e.g. obs, minute, hour, night, day, month, user)
+        spectral_model: `~gammapy.spectrum.models.SpectralModel`
+            spectral model
+        energy_range: `~astropy.units.Quantity`
+            true energy range to evaluate integrated flux (true energy)
+        binning: `~astropy.units.Quantity` 
+            bining 
+        """
+
+        # create time intervals
+        if binning == 'user':
+            t_intervals = binning
+        else:
+            t_intervals = self._get_time_intervals(t_start,
+                                                   t_stop,
+                                                   t_binning_type)
+
+        lc_flux = []
+        lc_flux_err = []
+        lc_tmin = []
+        lc_tmax = []
+            
+        # Loop on time intervals
+        for t_bin in t_intervals:
```

Comment:
I think it's possible to move all of the code processing one time interval into it's own function / method / class / whatever you like, no?

That would make the code easier to read and test and also a bit more re-usable, if it's possible to just compute one time bin.